kong/core/reports.lua

```diff
@@ -72,19 +72,27 @@ local function send_report(signal_type, t, host, port)
 
   local mutable_idx = _buffer_immutable_idx
 
+  local function f(k, v)
+    mutable_idx = mutable_idx + 1
+    _buffer[mutable_idx] = k .. "=" .. tostring(v)
```

Comment:
Any chance that instead of creating a function here (wrt. reference code), and giving it such an obscure name, we could simply build our loop below like so:
```lua
if k == "unique_id" or (k ~= "created_at" and sub(k, -2) ~= "id") then
  if type(v) == "function" then
    v = v()

  elseif type(v) == "table" then
    v = cjson.encode(v)
  end

  mutable_idx = mutable_idx + 1
  _buffer[mutable_idx] = k .. "=" .. tostring(v)
end
```

Reply:
Hmm, yep. For some reason I thought that reassigned `v` here would mean that the function would only be called once. Duh, we're in a `pairs` iterator, not touching `_ping_infos` directly. Will fix up.